pkg/nfd-topology-updater/nfd-topology-updater.go

```diff
@@ -225,11 +230,28 @@ func (w *nfdTopologyUpdater) Stop() {
 }
 
 func (w *nfdTopologyUpdater) updateNodeResourceTopology(zoneInfo v1alpha2.ZoneList, scanResponse resourcemonitor.ScanResponse, readKubeletConfig bool) error {
+	ownerRefs := []metav1.OwnerReference{}
+
+	ns, err := w.k8sClient.CoreV1().Namespaces().Get(context.TODO(), w.kubernetesNamespace, metav1.GetOptions{})
+	if err != nil {
+		klog.InfoS("Cannot set NodeResourceTopology owner reference: %w", err)
+	} else {
+		ownerRefs = []metav1.OwnerReference{
+			{
+				APIVersion: "v1",
+				Kind:       "Namespace",
+				Name:       ns.Name,
+				UID:        types.UID(ns.UID),
```

Comment:
One optimization I'd suggest is to cache the UID (e.g. `kubernetesNamespaceUID` similar `w.kubernetesNamespace`). Only get it once (or as long as it's empty) and then use the cached value as it's not going to change over the lifetime of the pod.

Reply:
We can probably just cache whole `OwnerReference`.

Reply:
@PiotrProkop @marquiz
Thanks, done.